go-controller/pkg/node/management-port.go

```diff
@@ -58,6 +65,55 @@ func (n *OvnNode) createManagementPort(hostSubnets []*net.IPNet, nodeAnnotator k
 		return err
 	}
 
+	if config.Gateway.Mode == config.GatewayModeLocal {
+		var gatewayIfAddrs []*net.IPNet
+		for _, hostSubnet := range hostSubnets {
+			// local gateway mode uses mp0 as default path for all ingress traffic into OVN
+			var nextHop *net.IPNet
+			if utilnet.IsIPv6CIDR(hostSubnet) {
+				nextHop = cfg.ipv6.ifAddr
+			} else {
+				nextHop = cfg.ipv4.ifAddr
+			}
+			gatewayIfAddrs = append(gatewayIfAddrs, nextHop)
+			stdout, stderr, err = util.RunOVNNbctl("--may-exist",
+				"--policy=src-ip", "lr-route-add", ovnClusterRouter,
+				hostSubnet.String(), nextHop.IP.String())
+			if err != nil {
```

Comment:
@dcbw @trozet this will not scale. We made a conscious choice to not make any ovn-nbctl calls from the node.

We don't have onv-nbctl in daemon mode running on the nodes and at scale these calls will timeout and never complete.

Reply:
It's just this one call only at node create time. I agree though don't want to set a precedent. However, in our annotations we only have:
                    k8s.ovn.org/node-mgmt-port-mac-address: 6a:31:e3:25:ce:b3
                    k8s.ovn.org/node-primary-ifaddr: {"ipv4":"172.18.0.4/16","ipv6":"fc00:f853:ccd:e793::4/64"}
                    k8s.ovn.org/node-subnets: {"default":"10.244.1.0/24"}

We don't actually store the mgmt ip address. So I could add that, which I'm kind of hesitant to do. Or I could assume it is always the nth address in the subnet, but that also seems not great. So I'm kind of inclined to just leave this as is for now unless we see some scale problems.

Reply:
> Or I could assume it is always the nth address in the subnet, but that also seems not great.

I feel like there's code somewhere else doing that already...

Reply:
I mean, we have `util.GetNodeManagementIfAddr(subnet *net.IPNet)`

Reply:
cool, I'll change it to use this in master node

Reply:
@trozet @girishmg agreed, this must be done on the master when the node logical network is set up. Not the node side.

Reply:
Also, why is this code in management-port.go, instead of in the gateway code where it belongs?

Reply:
because of the order of gateway init and mgmt init